### PR TITLE
Migrate gRPC tests to Java

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    while (!closer.isEmpty()) {
+      closer.poll().run();
+    }
+  }
+
+  @CartesianTest
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  void conversation(
+      @CartesianTest.Values(ints = {1, 2, 3}) int clientMessageCount,
+      @CartesianTest.Values(ints = {1, 2, 3}) int serverMessageCount)
```
Comment (trask) [inline instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java:60]: 👍

File: instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+                .collect(Collectors.toList()));
+
+    List<Consumer<EventDataAssert>> events = new ArrayList<>();
+    for (int i = 1; i <= clientMessageCount * serverMessageCount + clientMessageCount; i++) {
+      long messageId = i;
+      events.add(
+          event ->
+              event
+                  .hasName("message")
+                  .hasAttributesSatisfying(
```
Comment (trask) [inline instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java:155]: do we have hasAttributesSatisfyingExactly on event yet?

File: instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+                                    SemanticAttributes.RPC_GRPC_STATUS_CODE,
+                                    (long) Status.Code.OK.value()),
+                                equalTo(SemanticAttributes.NET_PEER_NAME, "localhost"),
+                                equalTo(SemanticAttributes.NET_PEER_PORT, (long) server.getPort()))
+                            .hasEventsSatisfyingExactly(events.toArray(new Consumer[0])),
+                    span ->
+                        span.hasName("example.Greeter/Conversation")
+                            .hasKind(SpanKind.SERVER)
+                            .hasParent(trace.getSpan(0))
+                            .hasAttributesSatisfyingExactly(
```
Comment (trask) [inline instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java:194]: this turned out nice 💯

**Review by trask — APPROVED** [review]

File: instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java (hunk 2 above)
Comment (anuraaga) [reply instrumentation/grpc-1.6/testing/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/AbstractGrpcStreamingTest.java:155]: Not yet unfortunately noticed the gap with this PR

**Review by Traviscal — COMMENTED** [review]
https://t.me/traviiscalls

**Pull request merged**